src/Functions/getServerSetting.cpp

```diff
@@ -0,0 +1,90 @@
+#include <Functions/IFunction.h>
+#include <Functions/FunctionFactory.h>
+#include <Functions/FunctionHelpers.h>
+#include <DataTypes/FieldToDataType.h>
+#include <Interpreters/convertFieldToType.h>
+#include <Interpreters/Context.h>
+#include <Core/Field.h>
+#include <Core/ServerSettings.h>
+
+
+namespace DB
+{
+namespace ErrorCodes
+{
+    extern const int ILLEGAL_TYPE_OF_ARGUMENT;
+    extern const int ILLEGAL_COLUMN;
+}
+
+namespace
+{
+
+class FunctionGetServerSetting : public IFunction, WithContext
+{
+public:
+    static constexpr auto name = "getServerSetting";
+
+    static FunctionPtr create(ContextPtr context_) { return std::make_shared<FunctionGetServerSetting>(context_); }
+    explicit FunctionGetServerSetting(ContextPtr context_) : WithContext(context_) {}
+
+    String getName() const override { return name; }
+
+    bool isDeterministic() const override { return false; }
+
+    bool isSuitableForShortCircuitArgumentsExecution(const DataTypesWithConstInfo & /*arguments*/) const override { return false; }
+
+    size_t getNumberOfArguments() const override { return 1 ; }
+
+    ColumnNumbers getArgumentsThatAreAlwaysConstant() const override { return {0}; }
+
+    DataTypePtr getReturnTypeImpl(const ColumnsWithTypeAndName & arguments) const override
+    {
+        auto value = getValue(arguments);
+        return applyVisitor(FieldToDataType{}, value);
+    }
+
+    ColumnPtr executeImpl(const ColumnsWithTypeAndName & arguments, const DataTypePtr & result_type, size_t input_rows_count) const override
+    {
+        auto value = getValue(arguments);
```

Comment:
Same comment as above, let's add a check for argument length and ensure that it's exactly 1.